driver/libexec/claw_f_lib.sh.in

```diff
@@ -754,22 +754,32 @@ function claw::process_dependencies() {
 
     ### Preprocess file ###
     base_file=${source_mod_file}
-    dep_basename="$(basename "${base_file%.*}")"
 
     # shellcheck disable=SC2154
     local file_pp
     file_pp="$(claw::get_pp_filename "${base_file}")"
 
+    claw::applyPreprocessorPass "${base_file}" "${file_pp}"
+
     # shellcheck disable=SC2154
     if [[ "${fpp_redirect}" == true ]]; then
+      local file_pp1
+      file_pp1="$(claw::get_pp1_filename "${base_file}")"
+
       # shellcheck disable=SC2086,SC2068
       ${OMNI_FPP_CMD} ${include_opt[@]} ${pp_add_opt[@]} ${OMNI_FPP_OPT} \
-        ${other_args[@]} "${base_file}" > "${file_pp}"
+        ${other_args[@]} "${file_pp}" > "${file_pp1}"
+
+      mv "${file_pp1}" "${file_pp}"
     else
+      local preprocessor_output
+      preprocessor_output="$(basename "${file_pp%.*}")"
+
       # shellcheck disable=SC2086,SC2068
       ${OMNI_FPP_CMD} ${include_opt[@]} ${pp_add_opt[@]} ${OMNI_FPP_OPT} \
-        ${other_args[@]} "${base_file}"
-      mv "${dep_basename}.i" "${file_pp}"
+        ${other_args[@]} "${file_pp1}"
```

Comment:
The line with `OMNI_FPP_CMD` is actually applying the preprocessor so what was not working on your side. Seems like we are applying the FPP twice now.

Reply:
@clementval [claw::applyPreprocessorPass](https://github.com/claw-project/claw-compiler/blob/cf83763900045058ea8373373dd4a4ce60235b21/driver/libexec/claw_f_lib.sh.in#L860-L886), which needs to be run, is not about running the preprocessor but about running grep.

Reply:
@skosukhin Right! I did not remember this :-)